packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -65,17 +89,17 @@ export class AttributionCollection<T> {
         this._length += other.length;
     }
 
-    public getAll(): { offset: number; key: T; }[] {
-        const results: { offset: number; key: T; }[] = [];
+    public getAll(): { offset: number; key: AttributionKey; }[] {
+        const results: { offset: number; key: AttributionKey; }[] = [];
         this.entries.map(({ key, data }) => {
-            results.push({ offset: key, key: data });
+            results.push({ offset: key, key: { id: "op", key: data } });
```

Comment:
i could see consumers wanting to walk the whole tree and gather all the sequence information. i worry this will be pretty costly with all the object allocations. i think it would be cheaper to have the caller specify the type of information they want, so "op" in this case. for now "op" could be the default, and we don't support anything else, so we can ignore it. also not totally clear how this fits into next steps.

Reply:
yeah, something like that was the planned direction I had for applications with either this or `getAtOffset`. We also do need a way to iterate all keys though in order to serialize correctly.

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/2921

resolving as implementation has no impact on API right now. We'll settle some more of the API next steps in merge-tree extensibility story.